datafusion/physical-expr/src/regex_expressions.rs

```diff
@@ -58,7 +59,7 @@ pub fn regexp_match<T: OffsetSizeTrait>(args: &[ArrayRef]) -> Result<ArrayRef> {
         2 => {
             let values = as_generic_string_array::<T>(&args[0])?;
             let regex = as_generic_string_array::<T>(&args[1])?;
-            compute::regexp_match(values, regex, None).map_err(|e| arrow_datafusion_err!(e))
+            _regexp_match(values, regex, None).map_err(|e| arrow_datafusion_err!(e))
```

Comment:
It seems we're also are not supporting the scalar version (I.e. regex being a literatal value)?

Reply:
I see you did some testing already https://github.com/apache/arrow-rs/pull/5235#issuecomment-1868071898

Reply:
I have bigger "plans" -- https://github.com/apache/arrow-datafusion/issues/8051 to specialize the constant arguments.

Reply:
Yea, for the ScalarUDF in DataFusion, we probably can rely on further optimization like #8051. Per kernel perspective, I will probably submit a Datum version later as mentioned https://github.com/apache/arrow-rs/pull/5235#issuecomment-1868071898.